tests/kafkatest/services/verifiable_producer.py

```diff
@@ -87,10 +109,10 @@ def _worker(self, idx, node):
                 with self.lock:
                     if data["name"] == "producer_send_error":
                         data["node"] = idx
-                        self.not_acked_values.append(int(data["value"]))
+                        self.not_acked_values.append(self.message_validator(data["value"]))
```

Comment:
@granders Very good point. Before, if the format is wrong, int() will throw an exception. As we just discussed, the consumer side will swallow the exception -- but if we got a message with unexpected format we should fail the test. I will change both validators (on both producer and consumer side) to throw an exception is format is wrong.